metadata-ingestion/src/datahub/ingestion/source/superset.py

```diff
@@ -519,6 +525,75 @@ def _process_dashboard(self, dashboard_data: Any) -> Iterable[MetadataWorkUnit]:
                     f"Dashboard '{dashboard_title}' (id: {dashboard_id}) filtered by dashboard_pattern"
                 )
                 return
+
+            if self.config.database_pattern != AllowDenyPattern.allow_all():
+                raw_position_data = dashboard_data.get("position_json", "{}")
+                position_data = (
+                    json.loads(raw_position_data)
+                    if raw_position_data is not None
+                    else {}
+                )
+
+                chart_ids = []
+                for key, value in position_data.items():
+                    if not key.startswith("CHART-"):
+                        continue
+                    chart_id = value.get("meta", {}).get("chartId")
+                    if chart_id:
+                        chart_ids.append(chart_id)
+
+                for chart_id in chart_ids:
+                    if chart_id in self.filtered_chart_to_database:
+                        database_name = self.filtered_chart_to_database[chart_id]
+                        if database_name and not self.config.database_pattern.allowed(
+                            database_name
+                        ):
+                            self.report.warning(
+                                f"Dashboard '{dashboard_title}' (id: {dashboard_id}) uses dataset filtered by database_pattern with database '{database_name}'"
+                            )
+                            return
+                    else:
+                        chart_response = self.session.get(
+                            f"{self.config.connect_uri}/api/v1/chart/{chart_id}",
+                            timeout=self.config.timeout,
+                        )
+
+                        if chart_response.status_code == 200:
+                            chart_data = chart_response.json().get("result", {})
+                            datasource_id = chart_data.get("datasource_id")
+
+                            if datasource_id:
+                                if datasource_id in self.filtered_dataset_to_database:
+                                    database_name = self.filtered_dataset_to_database[
+                                        datasource_id
+                                    ]
+                                else:
+                                    dataset_response = self.get_dataset_info(
+                                        datasource_id
+                                    )
+                                    database_name = (
+                                        dataset_response.get("result", {})
+                                        .get("database", {})
+                                        .get("database_name", "")
```

Comment:
```suggestion
                                        .get("database_name")
```

This will return `None` if missing.